addons/l10n_ar_withholding/wizards/l10n_ar_payment_register_withholding.py

```diff
@@ -20,15 +22,40 @@ class l10nArPaymentRegisterWithholding(models.TransientModel):
         'account.tax', check_company=True, required=True,
         domain="[('l10n_ar_withholding_payment_type', '=', parent.partner_type)]")
     withholding_sequence_id = fields.Many2one(related='tax_id.l10n_ar_withholding_sequence_id')
-    base_amount = fields.Monetary(required=True)
+    base_amount = fields.Monetary(required=True, compute='_compute_base_amount', store=True, readonly=False)
     amount = fields.Monetary(required=True, compute='_compute_amount', store=True, readonly=False)
 
     def _tax_compute_all_helper(self):
         self.ensure_one()
         # Computes the withholding tax amount provided a base and a tax
         # It is equivalent to: amount = self.base * self.tax_id.amount / 100
+
+        # if it is earnings withholding, then we accumulate the tax base for the period
+        if self.tax_id.l10n_ar_tax_type in ['earnings', 'earnings_scale']:
+            to_date = self.payment_register_id.payment_date or datetime.date.today()
+            from_date = to_date + relativedelta(day=1)
+            # We search for the payments in the same month of the same regimen and the same code.
+            # self.env['account.move.line']._check_company_domain(self.env.companies)
+            same_period_withholdings = abs(sum(self.env['account.move.line'].search([
```

Comment:
read_group? [balance:sum]  (grouped by l10n_ar_code e.g.)  As otherwise I don't think we really do something with these.  (so no need to have it in cache)

Reply:
```suggestion
            same_period_withholdings = self.env['account.move.line']._read_group(domain, ['l10n_ar_code'], ['balance:sum'])[0][1] 
```
Maybe this works.

Reply:
@jco-odoo it doesn´t work, I will try to make it work